src/peergos/server/sql/SqlSupplier.java

```diff
@@ -24,6 +28,43 @@ default String createTransactionsTableCommand() {
                 "tid varchar(64) not null, owner varchar(64) not null, hash varchar(64) not null);";
     }
 
+    String insertOrIgnoreCommand(String prefix, String suffix);
+
+    default String createUsageTablesCommand() {
+        return "CREATE TABLE IF NOT EXISTS users (" +
+                "id " + getSerialIdType() + " PRIMARY KEY NOT NULL," +
```

Comment:
If we generate the Id field in java code, then we do not need to do a select after an insert in order to get it from the db.

Reply:
We don't generate it in Java code

Reply:
This is just to handle the difference between sqlite and postgres syntax

Reply:
I meant - if we did generate the id fields in java code... Database generated sequence numbers are not as painless as they seem.

Reply:
But then you will be racing other connections to insert the id, with potentially many retries required. Doing it in the db guarantees a single call.